www/js/metrics.js

```diff
@@ -692,77 +699,82 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
+
+          twoWeeksAgoCarbon = FootprintHelper.readableFormat(twoWeeksAgoCarbonInt);
+          lastWeekCarbon    = twoWeeksAgoCarbon;
         }
       }
+      $scope.carbonData.lastWeekUserCarbon = lastWeekCarbon;
 
       var change = "";
-      console.log("Running calculation with "
-                    + (lastWeekCarbonInt[0] + lastWeekCarbonInt[1])
-                    + " and "
-                    + (twoWeeksAgoCarbonInt[0] + twoWeeksAgoCarbonInt[1]))
-      var calculation = (((lastWeekCarbonInt[0] + lastWeekCarbonInt[1]) / 2)
-                        / ((twoWeeksAgoCarbonInt[0] + twoWeeksAgoCarbonInt[1]) / 2))
-                        * 100 - 100;
+      console.log("Running calculation with " + lastWeekCarbonInt + " and " + twoWeeksAgoCarbonInt);
+      var calculation = (lastWeekCarbonInt/twoWeeksAgoCarbonInt) * 100 - 100;
 
       // TODO: Refactor this so that we can filter out bad values ahead of time
       // instead of having to work around it here
       if (isValidNumber(calculation)) {
-          if(lastWeekCarbonInt[0] > twoWeeksAgoCarbonInt[0]){
-            $scope.carbonData.change = " increase over a week";
-            $scope.carbonUp = true;
-            $scope.carbonDown = false;
-          } else {
-            $scope.carbonData.change = " decrease over a week"
-            $scope.carbonUp = false;
-            $scope.carbonDown = true;
-          }
-          $scope.carbonData.changeInPercentage = Math.abs(Math.round(calculation)) + "%"
+        if(lastWeekCarbonInt > twoWeeksAgoCarbonInt){
+          $scope.carbonData.change = $translate.instant('metrics.carbon-data-change-increase');
+          $scope.carbonUp = true;
+          $scope.carbonDown = false;
+        } else {
+          $scope.carbonData.change = $translate.instant('metrics.carbon-data-change-decrease');
+          $scope.carbonUp = false;
+          $scope.carbonDown = true;
+        }
+        $scope.carbonData.changeInPercentage = Math.abs(Math.round(calculation)) + "%"
+      }
+      else {
+        $scope.carbonData.change = "";
+        $scope.carbonData.changeInPercentage = "0%";
       }
    };
 
    $scope.fillFootprintAggVals = function(aggDistance) {
       if (aggDistance) {
         var aggrCarbonData = getAvgSummaryDataRaw(aggDistance, 'distance');
+
+        // TIAGO: please remove before PR, and turn into an issue?
+        // It can happen that the mode "ON_FOOT" has a NaN value.
+        // This may be due to the fact that the aggregated values are requested
+        // from the main e-mission server and not from our own.
         for (var i in aggrCarbonData) {
-          if (aggrCarbonData[i].key === "IN_VEHICLE") {
-            $scope.carbonData.aggrVehicleRange = FootprintHelper.getFootprintRaw(aggrCarbonData[i].values, aggrCarbonData[i].key);
-            $scope.carbonData.aggrCarbon = FootprintHelper.getFootprint(aggrCarbonData[i].values, aggrCarbonData[i].key);
+          if (isNaN(aggrCarbonData[i].values)) {
```

Comment:
This has been investigated a bit more using the master phone branch and turned into an issue.
[https://github.com/e-mission/e-mission-docs/issues/422](https://github.com/e-mission/e-mission-docs/issues/422)